src/js/tracks/text-track.js

```diff
@@ -262,9 +214,7 @@ class TextTrack extends EventTarget {
       tt.loaded_ = true;
     }
 
-    if (browser.IS_IE8) {
-      return tt;
-    }
+    return retval;
```

Comment:
This should also still check for IE8 and return retval then. As do all the tracks.